### ISSUE TITLE
Add the ability to rename tables

### ISSUE THREAD

**Issue by TT1103** [apache/pinot#9136]
This would be useful for managing multiple versions of tables and swapping which table is the latest version. 
Alternatively, adding views would be good too.

**Comment by shaneauerbach** [comment]
I agree that this would be very useful. Two common situations I face:
1. I build a big realtime table for users. Takes anywhere from a couple of hours to a week to build. They start using it. And then something happens and it stops ingesting new data. I need to rebuild the table as soon as possible and it has to have the same name because my users reference it by name in their APIs and I don't want to ask them all to change their APIs. So, on one hand, I need to delete the broken table as soon as possible to free up the name to create the new table. On the other hand, I don't want to delete the broken table because I'm using it to debug/triage how I broke it. If I could simply rename it to append _broken, I could start rebuilding the new one immediately, limiting downtime. Similarly, a view would make this easy.
2. I build a big realtime table for users. One user asks that I add a column. It's a good idea, and something that other users may benefit from also. But, again, I can't change the table name because users are referencing it, so currently I would take down the table and rebuild it with the new column. Would be much better if I could build the new table under a different name, then, once its built, take down the old version and give its name to the new version. Or, again, views would work too--probably a more elegant solution.

**Comment by Jackie-Jiang** [comment]
See #10712 for more discussions
